### PR TITLE
feat(sync): add confirmation dialog before sync execution

### PR DESCRIPTION
**Pull request by ryota-murakami**
## Summary

Previously, clicking Sync with no conflicts would execute immediately — creating symlinks without any user confirmation. Now a confirmation dialog appears showing exactly what will happen: how many skills, how many agents, how many new symlinks to create, and how many are already linked.

**Changes:**
- **New `SyncConfirmDialog` component** — Shows preview stats before execution. Opens when `toCreate > 0` and no conflicts exist. Follows the same pattern as `SyncConflictDialog`.
- **`shouldShowSyncConfirm` helper** — Pure function extracted to `lib/syncHelpers.ts` with 5 unit tests covering null, conflicts, already-synced, and boundary cases.
- **SourceCard refactored** — Removed inline auto-execute logic. Now delegates to the appropriate dialog via Redux `syncPreview` state.
- **Review fixes:** Clean up stale `syncPreview` on early-return paths (no skills / already synced). Handle partial sync failures with `toast.warning` in both `SyncConfirmDialog` and `SyncConflictDialog`.

## Test Coverage
- `syncHelpers.test.ts`: 5 unit tests for `shouldShowSyncConfirm` (null, toCreate > 0, toCreate === 0, conflicts exist, zero + no conflicts)
- Tests: 24 files, 266 tests all passing

## Pre-Landing Review
Pre-Landing Review: 2 informational issues — both fixed.
- `SourceCard.tsx:60` — syncPreview leak on early returns → fixed (dispatch `setSyncPreview(null)`)
- `SyncConfirmDialog.tsx:45` + `SyncConflictDialog.tsx:63` — partial failures silently ignored → fixed (`toast.warning` for `errors.length > 0`)

Adversarial review (Claude + Codex): TOCTOU window between preview and execute noted as architectural (low practical risk for desktop app). No critical findings.

## Plan Completion
Plan: 8/8 items DONE (confirm dialog, cancel, conflict path, all 4 edge cases).

## Scope Drift
Scope Check: CLEAN — delivered exactly what was planned, plus review-driven fixes for partial failure handling (applied to both dialogs for consistency).

## Test plan
- [x] All vitest tests pass (266 tests, 24 files)
- [x] TypeScript typecheck passes
- [x] Pre-landing review: 0 critical, 2 informational (both fixed)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a confirmation dialog for sync operations that previews changes (skills, agents, new symlinks) before execution.
  * Enhanced sync completion notifications with detailed error reporting when syncs complete with issues.

* **Tests**
  * Added validation tests for sync preview functionality.

### CODE REVIEW

File: src/renderer/src/components/sidebar/SyncConfirmDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+          toast.warning('Sync completed with errors', {
+            description: `Created ${created} symlinks, ${errors.length} failed`,
+          })
+        } else {
+          toast.success('Sync completed', {
+            description: `Created ${created} symlinks`,
+          })
+        }
+        refreshAllData(dispatch)
+      } else {
```
[the remark this answers is not in the document]
Comment (ryota-murakami) [reply src/renderer/src/components/sidebar/SyncConfirmDialog.tsx:57]: This is already handled by the Redux reducer. `executeSyncAction.fulfilled` at `uiSlice.ts:145-148` sets `state.syncPreview = null`, which causes `shouldShowSyncConfirm(syncPreview)` to return `false` and the dialog closes automatically. No component-level cleanup needed — the reducer is the source of truth for this state transition. Same pattern used in `SyncConflictDialog`.
